hivemind/averaging/allreduce.py

```diff
@@ -343,10 +343,13 @@ async def _accumulate_parts_streaming(self, stream: AsyncIterator[averaging_pb2.
                 stream,
                 max_prefetch=self.tensor_part_container.prefetch,
             ):
-                averaged_part = await self.tensor_part_reducer.accumulate_part(
-                    sender_index, part_index, tensor_part, weight=weight
-                )
-                part_index += 1
+                try:
+                    averaged_part = await self.tensor_part_reducer.accumulate_part(
+                        sender_index, part_index, tensor_part, weight=weight
+                    )
+                    part_index += 1
+                except BannedException:
+                    break  # sender was banned, we no longer need to aggregate it
```

Comment:
Let's print a debug log message here if it's not already printed anywhere else :)

Reply:
done